rex/resource.py

```diff
@@ -1151,10 +1182,88 @@ def _map_file_instances(h5_files):
 
         return h5_map
 
+    def _preflight_check(self):
+        """
+        Check time_index and coordinates accross files
+        """
+        time_index = None
+        lat_lon = None
+
+        bad_files = []
+        for file in self.h5_files:
+            with Resource(file) as f:
+                ti = f.time_index
+                ll = f.lat_lon
+                if time_index is None:
+                    time_index = ti.copy()
+                else:
+                    check = time_index.equals(ti)
+                    if not check:
+                        bad_files.append(file)
+
+                if lat_lon is None:
+                    lat_lon = ll.copy()
+                else:
+                    check = np.allclose(lat_lon, ll)
+                    if not check:
+                        bad_files.append(file)
+
+        bad_files = list(set(bad_files))
+        if bad_files:
+            msg = ("The following files' coordinates and time-index do not "
+                   "match:\n{}".format(bad_files))
+            raise ResourceRuntimeError(msg)
+
+    def close(self):
+        """
+        Close all h5py.File instances
+        """
+        for f in self._h5_map.values():
+            f.close()
+
+
+class MultiH5Path(MultiH5):
+    """
+    Class to handle multiple h5 file Resources derived from a path
+    """
+    def __init__(self, h5_path, prefix='', suffix='.h5', check_files=False):
+        """
+        Parameters
+        ----------
+        h5_path : str
+            Path to directory containing multi-file resource file sets.
+            Available formats:
+                /h5_dir/
+                /h5_dir/prefix*suffix
+        prefix : str
+            Prefix for resource .h5 files
+        suffix : str
+            Suffix for resource .h5 files
+        check_files : bool
+            Check to ensure files have the same coordinates and time_index
+        """
+        self.h5_dir, pre, suf = self.multi_file_args(h5_path)
+        if pre is None:
+            pre = prefix
+
+        if suf is None:
+            suf = suffix
+
+        h5_files = self._get_h5_files(self.h5_dir, prefix=pre, suffix=suf)
+        super().__init__(h5_files, check_files=check_files)
+
+    def __repr__(self):
+        msg = ("{} for {}:\n Contains {} files and {} datasets"
+               .format(self.__class__.__name__, self.h5_dir,
+                       len(self), len(self._dset_map)))
+
+        return msg
+
     @staticmethod
     def multi_file_args(h5_path):
```

Comment:
I think this method in combination with `_get_h5_files()` will fail if the h5_path is something simple like `prefix*.h5` (without a directory part). When I call `os.path.split('test.h5')` I get: `('', 'test.h5')` and then `os.listdir('')` throws a `FileNotFoundError`. I think I ran into this a while ago and didnt tell you (oops). Easy solution is to use `./prefix*.h5` but it would be nice if we didnt have to do that.

Reply:
ah, I never ran into that. I'll add a test and a fix

Reply:
Yeah it was just me poking around some files in ipython and isn't a common case, but easy to handle :)

Reply:
Check my fix, I just used os.path.abspath, I think it should work

Reply:
Effective, i like it!